stock_location_orderpoint/models/stock_location_orderpoint.py

```diff
@@ -0,0 +1,429 @@
+                qties_replenished_for_location = qties_replenished.setdefault(
+                    orderpoint.location_id, {}
+                )
+                qties_replenished_for_location.setdefault(product, 0)
+
+                qty_to_replenish = orderpoint._get_qty_to_replenish(
+                    product,
+                    qties_on_locations,
+                    qties_replenished_for_location[product],
+                )
+                if (
+                    float_compare(
+                        qty_to_replenish, 0, precision_rounding=product.uom_id.rounding
+                    )
+                    > 0
+                ):
+                    qties_to_replenish.setdefault(
+                        orderpoint,
+                        [],
+                    )
+                    qties_to_replenish[orderpoint].append((product, qty_to_replenish))
+                    qties_replenished_for_location[product] += qty_to_replenish
+        return qties_to_replenish
+
+    def __prepare_procurements(self, moves_by_location):
+        qties_to_replenish_by_orderpoint = self._get_qties_to_replenish(
+            moves_by_location
+        )
+        procurements = []
+        for orderpoint, qties_to_replenish in qties_to_replenish_by_orderpoint.items():
+            proc_vals = orderpoint._prepare_procurement_values()
+            for product, qty in qties_to_replenish:
+                date_planned = min(
+                    moves_by_location[orderpoint.location_id]
+                    .filtered(lambda move: move.product_id == product)
+                    .mapped("date")
+                )
+                procurements.append(
+                    orderpoint._prepare_procurement(
+                        product, qty, date_planned, proc_vals
+                    )
+                )
+        return procurements
+
+    def _prepare_procurements(self, products=False):
+        moves_by_location = self._find_potential_moves_to_replenish_by_location(
+            products
+        )
+        return self._sort_orderpoints().__prepare_procurements(moves_by_location)
+
+    def run_replenishment(self, products=False):
+        """Run the replenishment for all potential products or only a selection"""
+        procurements = self._prepare_procurements(products)
+        if not procurements:
+            return
+        self.env["procurement.group"].with_context(from_orderpoint=True).run(
+            procurements, raise_user_error=False
+        )
+        self._after_replenishment()
+
+    def _prepare_to_assign_replenishment_move_domain(self):
+        """Returns a domain which selects moves created by a replenishment"""
+        domain = [
+            ("state", "in", ["confirmed", "partially_available"]),
+            ("procure_method", "=", "make_to_stock"),
+        ]
+        orderpoint_domains = []
+        for orderpoint in self:
+            _domain = [("origin", "=", orderpoint.name)]
```

Comment:
I don't like this at all. You can not rely on a string name to search for records. For a reason X or Y, if the name has changed, this won't work.

You should use the location_orderpoint_id field instead.

So, you can add it before the loop `[(location_orderpoint_id', in, self.ids)]`

Reply:
So, then, is there a need to add the location_id as we are strict enough with the orderpoint id ?

Reply:
I've tested this and tests are passing (maybe some cases are not included ?):

```python
def _prepare_to_assign_replenishment_move_domain(self):
        """Returns a domain which selects moves created by a replenishment"""
        domain = [
            ("state", "in", ["confirmed", "partially_available"]),
            ("procure_method", "=", "make_to_stock"),
            ("location_orderpoint_id", "in", self.ids)
        ]
        return domain
```

Reply:
This should be fine, i implemented the location_orderpoint_id field later in the process. 
Thats why there was the origin search. Thx

Reply:
I don't agree on that as:

- Either you should add a unique constraint for the name too.
- 'This should be fine,' is not a jsutification. Could you elaborate a little bit more why you are searching on origin (maybe there is a good reason :smiley: )?